hw02-generics/src/main/java/com/atumra/CustomerService.java

```diff
@@ -0,0 +1,53 @@
+package com.atumra;
+
+import java.util.AbstractMap;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.Map.Entry;
+
+public class CustomerService {
+
+    private final TreeMap<Customer, String> treeMap = new TreeMap<>(Comparator.comparing(Customer::getScores));
```

Comment:
Переменные (поля) лучше не называть по их типу.
Имена должны отображать то, что хранится в переменных, а там Customer.

`treeMap` -> `customers`

```suggestion
    private final TreeMap<Customer, String> customers = new TreeMap<>(Comparator.comparing(Customer::getScores));
```